packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -289,160 +410,281 @@ export function useConnections({
       }
+      debug('connection closed', connectionId);
     },
-    [onConnected, persistOIDCTokens, saveConnectionInfo]
+    [connectionsManager]
   );
 
-  useEffect(() => {
-    if (connectionStorage.getAutoConnectInfo) {
-      void connect(
-        connectionStorage.getAutoConnectInfo.bind(connectionStorage),
-        false
-      ).catch(() => {
-        // noop, we're already logging in the connect method
-      });
-    }
-  }, [connectionStorage]);
-
-  const closeConnection = async (connectionId: string) => {
-    debug('closing connection with connectionId', connectionId);
-    log.info(
-      mongoLogId(1_001_000_313),
-      'Connection UI',
-      'Initiating disconnect attempt'
-    );
-    try {
-      await connectionsManager.closeConnection(connectionId);
-    } catch (error) {
-      log.error(
-        mongoLogId(1_001_000_314),
-        'Connection UI',
-        'Disconnect attempt failed',
-        {
-          error: (error as Error).message,
-        }
-      );
-    }
-    debug('connection closed', connectionId);
-  };
+  const createNewConnection = useCallback(() => {
+    dispatch({ type: 'new-connection' });
+  }, []);
 
-  const connect = async (
-    _connectionInfo:
-      | ConnectionInfo
-      | (() => Promise<ConnectionInfo | undefined>),
-    legacyShouldSaveConnectionInfo?: boolean
-  ) => {
-    const isAutoconnectAttempt = typeof _connectionInfo === 'function';
-    let connectionInfo: ConnectionInfo;
-
-    try {
-      if (typeof _connectionInfo === 'function') {
-        log.info(
-          mongoLogId(1_001_000_160),
-          'Connection Store',
-          'Performing automatic connection attempt'
-        );
-        const autoConnectInfo = await _connectionInfo();
-        if (!autoConnectInfo) {
-          return;
+  const editConnection = useCallback(
+    (connectionId: string) => {
+      const connectionInfo = getConnectionInfoById(connectionId);
+      if (connectionInfo) {
+        dispatch({ type: 'edit-connection', connectionInfo });
+      }
+    },
+    [getConnectionInfoById]
+  );
+
+  const cancelEditConnection = useCallback(() => {
+    dispatch({ type: 'cancel-edit-connection' });
+  }, []);
+
+  const duplicateConnection = useCallback(
+    async (
+      connectionId: string,
+      { autoDuplicate }: { autoDuplicate: boolean } = { autoDuplicate: false }
+    ) => {
+      const connectionInfo = getConnectionInfoById(connectionId);
+
+      if (!connectionInfo) {
+        return;
+      }
+
+      const findConnectionByFavoriteName = (name: string) => {
+        return findConnectionInfo((connectionInfo) => {
+          return connectionInfo.favorite?.name === name;
+        });
+      };
+
+      const duplicate: ConnectionInfo = {
+        ...cloneDeep(connectionInfo),
+        id: new UUID().toString(),
+      };
+
+      if (duplicate.favorite?.name) {
+        const copyFormat = duplicate.favorite?.name.match(/(.*)\s\(([0-9])+\)/); // title (2) -> [title (2), title, 2]
```

Comment:
Is this missing anchors? Otherwise we match something like `foo (2) bar` as well and lose everything after the closing parenthesis

```suggestion
        const copyFormat = duplicate.favorite?.name.match(/^(.*)\s\(([0-9])+\)$/); // title (2) -> [title (2), title, 2]
```

Reply:
Good catch! I'll fix this implementation and add a test (we can also probably reduce the complexity of the algorithm there by reducing to the highest number right away instead of looping over for every number 🤔)

Reply:
Updated in https://github.com/mongodb-js/compass/pull/6059/commits/ade3ada9978a3950ec2f0de12675b97b2252e08c, also fixed handling multi digit numbers